include/geom/elem.h

```diff
@@ -2053,6 +2062,17 @@ class Elem : public ReferenceCountedObject<Elem>,
     return ElemClass::side_nodes_map[s];
   }
 
+  /**
+   * Helper for overriding nodes_on_side in derived classes.
+  */
+  template <class ElemClass>
+  const unsigned int * _nodes_on_edge_ptr(const unsigned short s) const
+  {
+    static_assert(std::is_base_of<Elem, ElemClass>::value, "Not an Elem");
+    libmesh_assert_less(s, std::extent<decltype(ElemClass::edge_nodes_map)>::value);
```

Comment:
Should we throw in a static assert here to make sure that `std::extent<decltype(ElemClass::edge_nodes_map)>::value` is also consistent with `ElemClass::nodes_per_edge`?   (and similarly in `_nodes_on_side_ptr`)

Reply:
I do like this but unfortunately it won't work because in cases like `Edge`, we don't have an `edge_nodes_map`